README.md

```diff
@@ -488,8 +488,10 @@ oct2 = 0o755 # useful for Unix file permissions
 bin1 = 0b11010110
 ```
 
-64 bit (signed long) range expected (−9,223,372,036,854,775,808 to
-9,223,372,036,854,775,807).
+Arbitrary 64-bit signed integers (from −2^63 to 2^63−1) should be accepted and
+handled losslessly. If the result of parsing a value with unlimited precision
+would result in an integer that cannot be represented losslessly, an error must
+be thrown.
```

Comment:
This feels like a very complex sentence -- I understand what this means, but it needed a couple of readings and conscious thinking to figure it out.

I think we can drop the "unlimited precision" part of this sentence without losing any of the main benefits. :)

```suggestion
handled losslessly. If an integer that cannot be represented losslessly,
an error must be thrown.
```

Reply:
@pradyung I agree to an extend. Though as advocate for the devil, a very literal reading can then become: I parse an int with `Int64.TryParse` and get back `-2`. Fine, right?

Nope, because that function would give a 2's complement result, input could've been `2^63 + 2`. Hence I introduced the wording "with unlimited precision".

I would highly recommend that at some point we diverge into two documents: one for the general public, and one being the actual spec using precise spec wording. The second one intended for implementers, who are expected to be able to read such more strict documents.

Reply:
Hmmm... I'm pretty sure I'd consider `2^63 + 2` becoming `-2` as a case of "cannot be represented losslessly".

> I would highly recommend that at some point we diverge into two documents: one for the general public, and one being the actual spec using precise spec wording. The second one intended for implementers, who are expected to be able to read such more strict documents.

Perhaps? I understand the balance we're trying to strike here, and if it's definitely-not-possible, we should do this. I'm not convinced that it'd be a worthwhile exercise right now though. :)

Reply:
PS: I think the devil is pretty hunky-dory right now. It'll be fine if you don't advocate for them for a bit. :P

Reply:
> I'm not convinced that it'd be a worthwhile exercise right now though. :)

Agreed, I didn't mean "right now", but more for a vNext ;)